### ISSUE TITLE
TimeTravel can't handle enums

### ISSUE THREAD

**Issue by DSeeLP** [arkivanov/MVIKotlin#55]
TimeTravel can't serialize Enum implementing `JvmSerializable`

My Enum:
```kotlin
    enum class ConnectionStatus(val friendlyName: String): JvmSerializable {
        DISCONNECTED("Disconnected"),
        CONNECTING("Connecting"),
        CONNECTED("Connected"),
        AUTHENTICATED("Authenticated"),
    }
```

Stacktrace:
```
Exception in thread "AWT-EventQueue-0" java.lang.reflect.InaccessibleObjectException: Unable to make field private final java.lang.String java.lang.Enum.name accessible: module java.base does not "opens java.lang" to unnamed module @1aeda28e
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:178)
	at java.base/java.lang.reflect.Field.setAccessible(Field.java:172)
	at com.arkivanov.mvikotlin.timetravel.proto.internal.data.value.ValueParser$Companion.trySetAccessibleCompat(ValueParser.kt:271)
	at com.arkivanov.mvikotlin.timetravel.proto.internal.data.value.ValueParser$Companion.accessEachField(ValueParser.kt:335)
	at com.arkivanov.mvikotlin.timetravel.proto.internal.data.value.ValueParser$Companion.access$accessEachField(ValueParser.kt:252)
	at com.arkivanov.mvikotlin.timetravel.proto.internal.data.value.ValueParser.toTreeDefault(ValueParser.kt:209)
	at com.arkivanov.mvikotlin.timetravel.proto.internal.data.value.ValueParser.toTreeOther(ValueParser.kt:189)
	at com.arkivanov.mvikotlin.timetravel.proto.internal.data.value.ValueParser.valueOfObject(ValueParser.kt:56)
	at com.arkivanov.mvikotlin.timetravel.proto.internal.data.value.ValueParser.value(ValueParser.kt:23)
	at com.arkivanov.mvikotlin.timetravel.proto.internal.data.value.ValueParser.getValue(ValueParser.kt:249)
	at com.arkivanov.mvikotlin.timetravel.proto.internal.data.value.ValueParser.access$getValue(ValueParser.kt:7)
	at com.arkivanov.mvikotlin.timetravel.proto.internal.data.value.ValueParser$toTreeDefault$1.invoke(ValueParser.kt:210)
	at com.arkivanov.mvikotlin.timetravel.proto.internal.data.value.ValueParser$toTreeDefault$1.invoke(ValueParser.kt:209)
	at com.arkivanov.mvikotlin.timetravel.proto.internal.data.value.ValueParser$Companion.accessEachField(ValueParser.kt:286)
	at com.arkivanov.mvikotlin.timetravel.proto.internal.data.value.ValueParser$Companion.access$accessEachField(ValueParser.kt:252)
	at com.arkivanov.mvikotlin.timetravel.proto.internal.data.value.ValueParser.toTreeDefault(ValueParser.kt:209)
	at com.arkivanov.mvikotlin.timetravel.proto.internal.data.value.ValueParser.toTreeOther(ValueParser.kt:189)
	at com.arkivanov.mvikotlin.timetravel.proto.internal.data.value.ValueParser.valueOfObject(ValueParser.kt:56)
	at com.arkivanov.mvikotlin.timetravel.proto.internal.data.value.ValueParser.value(ValueParser.kt:23)
	at com.arkivanov.mvikotlin.timetravel.proto.internal.data.value.ValueParser.value$default(ValueParser.kt:21)
	at com.arkivanov.mvikotlin.timetravel.proto.internal.data.value.ValueParser.parseValue(ValueParser.kt:15)
	at com.arkivanov.mvikotlin.timetravel.server.TimeTravelServerImpl.analyzeEvent(TimeTravelServerImpl.kt:116)
	at com.arkivanov.mvikotlin.timetravel.server.TimeTravelServerImpl.access$analyzeEvent(TimeTravelServerImpl.kt:19)
	at com.arkivanov.mvikotlin.timetravel.server.TimeTravelServerImpl$onCommandReceived$$inlined$runOnMainThreadIfNotDisposed$1.invoke(TimeTravelServerImpl.kt:181)
	at com.arkivanov.mvikotlin.timetravel.server.TimeTravelServerImpl$onCommandReceived$$inlined$runOnMainThreadIfNotDisposed$1.invoke(TimeTravelServerImpl.kt:153)
	at io.github.dseelp.robotsystem.frontend.desktop.MainKt$timeTravelServer$1.invoke$lambda-0(Main.kt:21)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:771)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:716)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:741)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Comment by arkivanov** [comment]
Thanks for raising, I will fix it. Btw, `JvmSerializable` is not required for parsing, it's only for export/import events.

**Comment by DSeeLP** [comment]
I also have a suggestion, catch the exception and rethrow a custom exception with it as cause and a message that explains the reason behind this error and maybe suggests adding the `@Transient` annotation.

**Comment by arkivanov** [comment]
Usually it's not possible to know in advance what kind of exceptions could be thrown, especially with the lack of checked exceptions in Kotlin. This particular exception (`InaccessibleObjectException`) is not even [documented](https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/AccessibleObject.html#setAccessible-boolean-) - there is only `SecurityException` mentioned, which is already handled.

Also parsing objects for time traveling shouldn't throw any exceptions, it should just skip properties that are not possible to parse. I'm thinking that it's may not be needed to check the accessibility of fields at all, so this condition may go away.

**Comment by arkivanov** [comment]
Fixed in version `3.0.2`.
